### ISSUE TITLE
Thoughts of defining `take!(::Response)`?

### ISSUE THREAD

**Issue by malmaud** [JuliaWeb/HTTP.jl#30]
I wonder if it would be a good idea to essentially implement the `IOBuffer` API from Julia .6 for `Response` objects.

**Comment by quinnj** [comment]
like
```julia
take!(::Type{String}, Response)
take!(Response) = take!(Vector{UInt8}, Response)
```
?

I really like that; it seems much less ad-hoc than the current `string` and `bytes` methods I threw in there. It also helps with indicating that you're actually _consuming_ the response, since you actually are consuming the data in teh underlying `FIFOBuffer`.

**Comment by quinnj** [comment]
Implemented here #31
